### PR TITLE
1400 geolocation map mouse click improvements

### PR DESCRIPTION
**Pull request by kellykelly3**

### CODE REVIEW

**Comment by kellykelly3** [comment]
I am not going to address the remaining codacy issues.  I did not write the search function that have complexity and would take some careful thought to simplify.  I would rather not take that risk.  

Codacy is also complaining about the search function taking a TreeSet instead of a Set, again I did not write that function.  search uses a function that is specific to TreeSet and not in the Set interface.  So I think we should let that one stand.

File: Core/src/org/sleuthkit/autopsy/geolocation/KdTree.java [hunk 1]
```diff
@@ -247,258 +163,208 @@ public boolean contains(T value) {
...
-            list.addAll(getTree(root.greater));
-        }
-
-        return list;
-    }
-
-    /** 
-     * Searches the K nearest neighbor.
+     * @param numNeighbors Number of neighbors to retrieve. Can return more than
+     *                     K, if last nodes are equal distances.
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/geolocation/KdTree.java]: "can return more than numNeighbors"

File: Core/src/org/sleuthkit/autopsy/geolocation/KdTree.java [hunk 2]
```diff
@@ -507,173 +373,258 @@ public EuclideanComparator(XYZPoint point) {
...
-            this.z = 0;
+        public XYZPoint(Double latitude, Double longitude) {
+            x = latitude;
+            y = longitude;
+            z = 0;
         }
 
         /**
-         * Default constructor
+         * Returns the x\latitude value for the point.
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/geolocation/KdTree.java]: Is that a backslash for a reason? Should it simply be x/latitude? makes it hard to read.

File: Core/src/org/sleuthkit/autopsy/geolocation/KdTree.java [hunk 3]
```diff
@@ -507,173 +373,258 @@ public EuclideanComparator(XYZPoint point) {
...
          *
-         * @param x
-         * @param y
-         * @param z
+         * @return x\Latitude value for the point
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/geolocation/KdTree.java]: same comment regarding backslash.

File: Core/src/org/sleuthkit/autopsy/geolocation/MapPanel.java [hunk 4]
```diff
@@ -434,16 +439,16 @@ private Point getLocationForDetailsPopup() {
...
+            return new ArrayList<>();
         }
 
         // Convert the mouse click location to latitude & longitude
-        GeoPosition geopos = mapViewer.getTileFactory().pixelToGeo(mouseClickPoint, mapViewer.getZoom());
+        GeoPosition geopos = mapViewer.convertPointToGeoPosition(clickPoint);
 
         // Get the 5 nearest neightbors to the point
-        Collection<MapWaypoint> waypoints = waypointTree.nearestNeighbourSearch(10, MapWaypoint.getDummyWaypoint(geopos));
+        Collection<MapWaypoint> waypoints = waypointTree.nearestNeighbourSearch(1, MapWaypoint.getDummyWaypoint(geopos));
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/geolocation/MapPanel.java]: Are we getting 5 nbrs or 1? comment is inconsistent with actual param.

**Pull request merged**